python/runtime/feature_derivation/feature_column.py

```diff
@@ -0,0 +1,512 @@
+        Get the underlying FieldDesc object list that the NumericColumn
+        object holds.
+
+        Returns:
+            A list of the FieldDesc objects, whose length is 1.
+        """
+        return [self.field_desc]
+
+    def apply_to(self, field_desc):
+        """
+        Create a new NumericColumn object that holds the given
+        FieldDesc object.
+
+        Args:
+            field_desc (FieldDesc): the given FieldDesc object.
+
+        Returns:
+            A new NumericColumn object which holds the given
+            FieldDesc object.
+        """
+        return NumericColumn(field_desc)
+
+
+class BucketColumn(CategoryColumn):
+    """
+    BucketColumn represents a bucketized feature column.
+
+    Args:
+        source_column (NumericColumn): the underlying NumericColumn object.
+        boundaries (list[int|float]): the boundaries of the buckets.
+    """
+    def __init__(self, source_column, boundaries):
+        assert isinstance(
+            source_column,
+            NumericColumn), "source_column of BUCKET must be of numeric type"
+        self.source_column = source_column
+        self.boundaries = boundaries
+
+    def get_field_desc(self):
+        """
+        Get the underlying FieldDesc object list that the BucketColumn
+        object holds.
+
+        Returns:
+            A list of the FieldDesc objects, whose length is 1.
+        """
+        return self.source_column.get_field_desc()
+
+    def apply_to(self, field_desc):
+        """
+        Create a new BucketColumn object that holds the given
+        FieldDesc object.
+
+        Args:
+            field_desc (FieldDesc): the given FieldDesc object.
+
+        Returns:
+            A new BucketColumn object which holds the given
+            FieldDesc object.
+        """
+        source_column = self.source_column.apply_to(field_desc)
+        return BucketColumn(source_column, self.boundaries)
+
+    def num_class(self):
+        """
+        Get the class number of the BucketColumn object.
+
+        Returns:
+            An integer which is equal to len(boundaries) + 1.
+        """
+        return len(self.boundaries) + 1
+
+
+class CategoryIDColumn(CategoryColumn):
+    """
+    CategoryIDColumn represents a categorical id feature column.
+
+    Args:
+        field_desc (FieldDesc): the underlying FieldDesc object.
+        bucket_size (int): the bucket size.
+    """
+    def __init__(self, field_desc, bucket_size):
+        assert isinstance(field_desc, FieldDesc)
+        self.field_desc = field_desc
+        self.bucket_size = bucket_size
+
+    def get_field_desc(self):
+        """
+        Get the underlying FieldDesc object list that the CategoryIDColumn
+        object holds.
+
+        Returns:
+            A list of the FieldDesc objects, whose length is 1.
+        """
+        return [self.field_desc]
+
+    def apply_to(self, field_desc):
+        """
+        Create a new CategoryIDColumn object that holds the given
```

Comment:
Maybe we can only keep the base class doc string?

Reply:
Great idea! Done.